### ISSUE TITLE
The error in sysrepoctl installation of o-ran-processing-element.yang

### ISSUE THREAD

**Issue by vancouversun** [sysrepo/sysrepo#3022]
I am installing O-RAN specification Yang module (O-RAN.WG4.MP-YANGs-v05.00).
When I'm trying to install o-ran-processing-element.yang, The installation is failed.
This is the error msg.

$ sysrepoctl -i o-ran-processing-element.yang -v3
```
[INF] Connection 74 created.
[ERR] Not found node "diffserv-markings" in path. (Schema location "/o-ran-processing-element:processing-elements/enhanced-uplane-mapping/uplane-mapping/up-markings/ipv4/upv4-dscp-name".)
[ERR] Target of leafref "upv4-dscp-name" cannot be referenced because it is disabled. (Schema location "/o-ran-processing-element:processing-elements/enhanced-uplane-mapping/uplane-mapping/up-markings/ipv4/upv4-dscp-name".)
sysrepoctl error: Failed to install modules (libyang error)
[INF] No datastore changes to apply.
```
diffserv-markings is defined in o-ran-interfaces.yang that I've already installed.

This is the partial o-ran-interfaces.yang
```yang
grouping dscp-marking {
    description
      "Configuration data for CU Plane DSCP marking";
    container diffserv-markings {
      description
        "DSCP Configuration";
      leaf u-plane-marking {
        type inet:dscp;
        default 46;
        description
          "Marking used for default u-plane flows.
          46 represents expedited forwarding";
      }
```
This is the installed module list
```
Module Name                        | Revision   | Flags | Owner     | Startup Perms | Submodules | Features
-----------------------------------------------------------------------------------------------------------
example-list                       |            | I     | root:root | 600           |            |
iana-hardware                      | 2018-03-13 | I     | root:root | 600           |            |
iana-if-type                       | 2017-01-19 | I     | root:root | 600           |            |
ietf-crypto-types                  | 2019-04-29 | I     | root:root | 600           |            |
ietf-datastores                    | 2018-02-14 | I     | root:root | 666           |            |
ietf-dhcpv6-types                  | 2018-09-04 | I     | root:root | 600           |            |
ietf-hardware                      | 2018-03-13 | I     | root:root | 600           |            |
ietf-inet-types                    | 2013-07-15 | I     | root:root | 444           |            |
ietf-interfaces                    | 2018-02-20 | I     | root:root | 600           |            |
ietf-ip                            | 2018-02-22 | I     | root:root | 600           |            |
ietf-netconf                       | 2013-09-29 | I     | root:root | 666           |            |
ietf-netconf-acm                   | 2018-02-14 | I     | root:root | 666           |            |
ietf-netconf-monitoring            | 2010-10-04 | I     | root:root | 600           |            |
ietf-netconf-notifications         | 2012-02-06 | I     | root:root | 666           |            |
ietf-netconf-with-defaults         | 2011-06-01 | I     | root:root | 666           |            |
ietf-network-instance              | 2019-01-21 | I     | root:root | 600           |            |
ietf-origin                        | 2018-02-14 | I     | root:root | 666           |            |
ietf-restconf                      | 2017-01-26 | I     | root:root | 600           |            |
ietf-subscribed-notifications      | 2019-09-09 | I     | root:root | 600           |            |
ietf-yang-library                  | 2019-01-04 | I     | root:root | 666           |            |
ietf-yang-metadata                 | 2016-08-05 | i     |           |               |            |
ietf-yang-schema-mount             | 2019-01-14 | I     | root:root | 666           |            |
ietf-yang-structure-ext            | 2020-06-17 | i     |           |               |            |
ietf-yang-types                    | 2013-07-15 | I     | root:root | 444           |            |
o-ran-ald                          | 2019-07-03 | I     | root:root | 600           |            |
o-ran-ald-port                     | 2019-07-03 | I     | root:root | 600           |            |
o-ran-antenna-calibration          | 2020-12-10 | I     | root:root | 600           |            |
o-ran-compression-factors          | 2020-08-10 | I     | root:root | 600           |            |
o-ran-delay-management             | 2020-08-10 | I     | root:root | 600           |            |
o-ran-dhcp                         | 2020-12-10 | I     | root:root | 600           |            |
o-ran-ethernet-forwarding          | 2020-04-17 | I     | root:root | 600           |            |
o-ran-externalio                   | 2019-07-03 | I     | root:root | 600           |            |
o-ran-fan                          | 2019-07-03 | I     | root:root | 600           |            |
o-ran-file-management              | 2019-07-03 | I     | root:root | 600           |            |
o-ran-fm                           | 2019-02-04 | I     | root:root | 600           |            |
o-ran-hardware                     | 2020-12-10 | I     | root:root | 600           |            |
o-ran-interfaces                   | 2020-12-10 | I     | root:root | 600           |            |
o-ran-laa                          | 2019-07-03 | I     | root:root | 600           |            |
o-ran-laa-operations               | 2019-07-03 | I     | root:root | 600           |            |
o-ran-lbm                          | 2019-02-04 | I     | root:root | 600           |            |
o-ran-module-cap                   | 2020-12-10 | I     | root:root | 600           |            |
o-ran-mplane-int                   | 2019-07-03 | I     | root:root | 600           |            |
o-ran-operations                   | 2020-12-10 | I     | root:root | 600           |            |
o-ran-software-management          | 2019-07-03 | I     | root:root | 600           |            |
o-ran-supervision                  | 2020-12-10 | I     | root:root | 600           |            |
o-ran-sync                         | 2020-08-10 | I     | root:root | 600           |            |
o-ran-trace                        | 2019-07-03 | I     | root:root | 600           |            |
o-ran-transceiver                  | 2019-07-03 | I     | root:root | 600           |            |
o-ran-troubleshooting              | 2019-02-04 | I     | root:root | 600           |            |
o-ran-udp-echo                     | 2019-02-04 | I     | root:root | 600           |            |
o-ran-usermgmt                     | 2020-12-10 | I     | root:root | 600           |            |
o-ran-ves-subscribed-notifications | 2020-12-10 | I     | root:root | 600           |            |
o-ran-wg4-features                 | 2020-12-10 | I     | root:root | 600           |            |
sysrepo-monitoring                 | 2022-08-19 | I     | root:root | 666           |            |
sysrepo-plugind                    | 2022-08-26 | I     | root:root | 666           |            |
yang                               | 2022-06-16 | I     | root:root | 666           |            |
```

**Comment by michalvasko** [comment]
The error is caused by the module not being compliant with the [YANG spec](https://www.rfc-editor.org/rfc/rfc7950.html#section-9.9) (last paragraph). If you think it is, please provide the YANG module and I will check it.

**Comment by michalvasko** [comment]
I am not able to load the module without its imports `o-ran-usermgmt` and `o-ran-wg4-features`, some revisions of these modules I found online seem to be too old and differ from the ones you are using. Please provide them but I think, based on the extensive usage of if-features that the error is correct and the YANG modules are not.

**Comment by vancouversun** [comment]
Please use this URL to download whole o-ran yang models that I am using.
https://orandownloadsweb.azurewebsites.net/download?id=46

**Comment by michalvasko** [comment]
Thanks, so the leafref `/o-ran-processing-element:processing-elements/enhanced-uplane-mapping/uplane-mapping/up-markings/ipv4/upv4-dscp-name` target `/if:interfaces/if:interface/ip:ipv4/o-ran-int:diffserv-markings` is conditional based on the feature `UDPIP-BASED-CU-PLANE`, the leafref itself has no `if-features`. That is in direct conflict with the paragraph I referenced before making the YANG `o-ran-processing-element` invalid.

**Comment by vancouversun** [comment]
Thanks for the reply. 
Is there a way to fix it to install o-ran-processing-element.yang?

**Comment by michalvasko** [comment]
Well, yes, just get rid of the problem I have mentioned but you will have to use a modified YANG module.

**Comment by peigaoxu** [comment]
I encountered the same error today while installing o-ran-processing-element.yang, and the current environment is the latest version deployed.
Error message:
 

> sysrepoctl -i /yangfile/o-ran-processing-element.yang -s /yangfile -o root -g root -v3

[INF] Connection 82 created.
[ERR] Not found node "diffserv-markings" in path. (Schema location "/o-ran-processing-element:processing-elements/enhanced-uplane-mapping/uplane-mapping/up-markings/ipv4/upv4-dscp-name".)
[ERR] Target of leafref "upv4-dscp-name" cannot be referenced because it is disabled. (Schema location "/o-ran-processing-element:processing-elements/enhanced-uplane-mapping/uplane-mapping/up-markings/ipv4/upv4-dscp-name".)
sysrepoctl error: Failed to install modules (libyang error)

> sysrepoctl -c o-ran-interfaces -e UDPIP-BASED-CU-PLANE -e ALIASMAC-BASED-CU-PLANE

After I enabled the UDPIP-BASED-CU-PLANE and ALIASMAC-BASED-CU-PLANE features of o-ran-interfaces, the installation was successful....
